head.go

```diff
@@ -953,6 +974,15 @@ func (h *Head) gc() {
 	// Remove deleted series IDs from the postings lists.
 	h.postings.Delete(deleted)
 
+	if h.wal != nil {
+		_, last, _ := h.wal.Segments()
+		h.deletedMtx.Lock()
+		for ref := range deleted {
```

Comment:
I think it would be worth including a comment here:
```suggestion
                // Keep series records until we're past segment 'last' because the WAL will still have
                // samples records with this ref ID. If we didn't keep these series records then on start up
                // when we replay the WAL, or any other code that reads the WAL, wouldn't be able to use
                // those samples since we would have no labels for that ref ID.
		for ref := range deleted {
```

Reply:
Done